From d32b205a018b36259076adb7e0db01c1529d38ba Mon Sep 17 00:00:00 2001
From: Felix Fietkau <nbd@nbd.name>
Date: Thu, 16 Mar 2023 19:47:26 +0530
Subject: [PATCH 127/500] OpenWrt: 430-mtk-bmt-support.patch

mtd/nand: add MediaTek NAND bad block managment table

Signed-off-by: Felix Fietkau <nbd@nbd.name>

Change-Id: Ia15153dfd8000d37bc9b6a3e033d8485bc705b38
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit af6fb742b068049a3e8e958f65b19b4ce6a1368f)
---
 drivers/mtd/nand/Kconfig  | 4 ++++
 drivers/mtd/nand/Makefile | 1 +
 2 files changed, 5 insertions(+)

diff --git a/drivers/mtd/nand/Kconfig b/drivers/mtd/nand/Kconfig
index 5b0c2c95f10c..e775c64e355d 100644
--- a/drivers/mtd/nand/Kconfig
+++ b/drivers/mtd/nand/Kconfig
@@ -46,6 +46,10 @@ config MTD_NAND_ECC_SW_BCH
 	  ECC codes. They are used with NAND devices requiring more than 1 bit
 	  of error correction.
 
+config MTD_NAND_MTK_BMT
+	bool "Support MediaTek NAND Bad-block Management Table"
+	default n
+
 config MTD_NAND_ECC_MXIC
 	bool "Macronix external hardware ECC engine"
 	depends on HAS_IOMEM
diff --git a/drivers/mtd/nand/Makefile b/drivers/mtd/nand/Makefile
index 19e1291ac4d5..fcd41139e468 100644
--- a/drivers/mtd/nand/Makefile
+++ b/drivers/mtd/nand/Makefile
@@ -3,6 +3,7 @@
 nandcore-objs := core.o bbt.o
 obj-$(CONFIG_MTD_NAND_CORE) += nandcore.o
 obj-$(CONFIG_MTD_NAND_ECC_MEDIATEK) += ecc-mtk.o
+obj-$(CONFIG_MTD_NAND_MTK_BMT)	+= mtk_bmt.o mtk_bmt_v2.o mtk_bmt_bbt.o mtk_bmt_nmbm.o
 
 obj-y	+= onenand/
 obj-y	+= raw/
-- 
2.34.1

